conda_forge_tick/migrators.xsh

```diff
@@ -0,0 +1,231 @@
+            if not source_url:
+                attrs['bad'] = '{}: missing url\n'.format($PROJECT)
+                return False
+            if isinstance(source_url, list):
+                for url in source_url:
+                    if 'Archive' not in url:
+                        source_url = url
+                        break
+            if 'cran.r-project.org/src/contrib' in source_url:
+                $VERSION = $VERSION.replace('_', '-')
+
+        # now, update the feedstock to the new version
+        source_url = eval_version(source_url)
+        try:
+            hash = hash_url(source_url, hash_type)
+        except urllib.error.HTTPError:
+            attrs['bad'] = '{}: hash failed at {}\n'.format(
+                meta_yaml.get('package', {}).get('name', 'UNKOWN'), source_url)
+            return False
+
+        patterns += tuple(more_patterns)
+        with indir(recipe_dir), ${...}.swap(HASH_TYPE=hash_type, HASH=hash, SOURCE_URL=source_url):
+            for f, p, n in patterns:
+                p = eval_version(p)
+                n = eval_version(n)
+                replace_in_file(p, n, f)
+        return True
+
+    def pr_body(self):
+        pred = [(name, $SUBGRAPH.node[name]['new_version'])
+                for name in list($SUBGRAPH.predecessors($NODE))]
+        body = super().pr_body()
+        body = body.format('Notes and instructions for merging this PR:\n'
+            '1. Please check that the dependencies have not changed. \n'
+            '2. Please merge the PR only after the tests have passed. \n'
+            "3. Feel free to push to the bot's branch to update this PR if needed. \n"
+            "4. The bot will almost always only open one PR per version. \n\n")
+        # Statement here
+        template = ('|{name}|{new_version}|[![Anaconda-Server Badge]'
+                    '(https://img.shields.io/conda/vn/conda-forge/{name}.svg)]'
+                    '(https://anaconda.org/conda-forge/{name})|\n')
+        if len(pred) > 0:
+            body += ('\n\nHere is a list of all the pending dependencies (and their '
+                     'versions) for this repo. '
+                     'Please double check all dependencies before merging.\n\n')
+            # Only add the header row if we have content.
+            # Otherwise the rendered table in the github comment
+            # is empty which is confusing
+            body += ('| Name | Upstream Version | Current Version |\n'
+                     '|:----:|:----------------:|:---------------:|\n')
+        for p in pred:
+            body += template.format(name=p[0], new_version=p[1])
+        return body
+
+    def commit_message(self):
+        return "updated v" + $VERSION
+
+
+class JS(Migrator):
+    """Migrator for JavaScript syntax"""
+    patterns = [('meta.yaml', '  script: *',
+                 '''  script: |
+                        tgz=$(npm pack)
+                        npm install -g $tgz'''), ]
```

Comment:
This renders as:

```
scopatz@artemis ~ $ print('''  script: |
`·.,¸,.·*¯`·.,¸,.·*                        tgz=$(npm pack)
`·.,¸,.·*¯`·.,¸,.·*                        npm install -g $tgz'''
`·.,¸,.·*¯`·.,¸,.·*                        )
`·.,¸,.·*¯`·.,¸,.·*                        
  script: |
                       tgz=$(npm pack)
                       npm install -g $tgz
```

In order to get the whitespace correct I think you want, 

```
('  script: |\n'
 '     tgz=$(npm pack)\n'
 '    npm install -g $tgz')
```